src/Propel/Runtime/Util/PropelDateTime.php

```diff
@@ -73,31 +73,41 @@ protected static function isTimestamp($value): bool
      *
      * @throws \InvalidArgumentException
      *
-     * @return \DateTime
+     * @return DateTimeInterface An instance of $dateTimeClass
      */
-    public static function createHighPrecision(?string $time = null): DateTime
+    public static function createHighPrecision(?string $time = null, string $dateTimeClass = 'DateTime'): DateTimeInterface
     {
-        $dateTime = DateTime::createFromFormat('U.u', $time ?: self::getMicrotime());
+        $dateTime = $dateTimeClass::createFromFormat('U.u', $time ?: self::getMicrotime());
```

Comment:
can we check that this provided class is an instance of DateTimeInterface in order to make sure that it has createFromFormat method?

Reply:
DateTimeInterface::createFromFormat doesn't exist apparently, so I added checks for both classes individually. 
Here the second parameter is also a string (and not DateTime::class), but other methods inside PropelDateTime use that too so I'd rather not mix updating that in this PR.